helper/linking.js

```diff
@@ -12,10 +12,13 @@ const linking = {
     },
     subscribe(listener) {
         const onReceiveURL = ({ url }) => listener(url);
-        Linking.addEventListener('url', onReceiveURL);
-        return () => {
-            Linking.removeEventListener('url', onReceiveURL);
-        };
+
+    const subscription = Linking.addEventListener('url', onReceiveURL);
+
+    return () => {
+
+        subscription.remove();
+    };
     },
```

Comment:
why do we need to change this file?

Reply:
In Expo SDK 51, the Linking API has been deprecated and replaced with the Linking API from expo-linking package. The addEventListener method is no longer available. Instead, should use the addLinkingListener method.